src/components/AppWrapper.tsx

```diff
@@ -0,0 +1,217 @@
+                }
+                draft.splice(row, 1)
+            })
+        } else {
+            infographicsCopy = produce(infographics, draft => {
+                if (column === draft[row].sections.length - 1) {
+                    draft[row].sections[column - 1].widthPercent += section.widthPercent
+                } else {
+                    draft[row].sections[column + 1].widthPercent += section.widthPercent
+                }
+                for (let i = column + 1; i < draft[row].sections.length; ++i) {
+                    --draft[row].sections[i].position.x
+                    --draft[row].sections[i].column.start
+                    --draft[row].sections[i].column.end
+                }
+                draft[row].sections.splice(column, 1)
+            })
+        }
+
+        this.setState(produce(this.state, draftState => {
+            draftState.infographicsDetails = {
+                widthPx,
+                infographics: infographicsCopy,
+            }
+        }))
+    };
+
+    private handleDivideSection = (section: IInfographicsSection) => {
+        const {infographicsDetails, lastUsedId: id} = this.state
+        const {infographics, widthPx} = infographicsDetails
+        const {x: column, y: row} = section.position
+        const {end: colEnd} = section.column
+
+        const newSectionWidth: number = section.widthPercent / 2
+
+        const dividedSectionUpdated = produce(section, draft => {
+            draft.widthPercent = newSectionWidth
+        })
+
+        const newSection: IInfographicsSection = {
+            id,
+            position: {
+                x: column + 1,
+                y: row,
+            },
+            column: {
+                start: colEnd,
+                end: colEnd + 1,
+            },
+            widthPercent: newSectionWidth,
+            isActive: false,
+        }
+
+        const infographicsUpdated = produce(infographics, draft => {
+            draft[row].sections[column] = dividedSectionUpdated
+            draft[row].sections.splice(column + 1, 0, newSection)
+            ++draft[row].columns
+            for (let i = column + 2; i < draft[row].sections.length; ++i) {
+                ++draft[row].sections[i].position.x
+                ++draft[row].sections[i].column.start
+                ++draft[row].sections[i].column.end
+            }
+        })
+
+        this.setState(produce(this.state, draftState => {
+            draftState.infographicsDetails = {
+                widthPx,
+                infographics: infographicsUpdated,
+            }
+            ++draftState.lastUsedId
+            draftState.selectedSection = dividedSectionUpdated
+        }))
+    }
+
+    private handleSetWidth = (widthPx: number) => {
+        const {infographicsDetails} = this.state
+        const {infographics} = infographicsDetails
+
+        this.setState(produce(this.state, draftState => {
+            draftState.infographicsDetails = {
+                widthPx,
```

Comment:
`widthPx` je ocividne jen jedna a `infographics` je pole, vytahnul bych to z `infographicsDetails` ven, aby jsi nemusela resit oboji a stacilo updatnout `width`